api/validation/etcd.go

```diff
@@ -66,8 +66,6 @@ func ValidateEtcdSpecUpdate(new, old *v1alpha1.EtcdSpec, deletionTimestampSet bo
 		return allErrs
 	}
 
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(new.Backup.Store, old.Backup.Store, path.Child("backup.store"))...)
-
```

Comment:
This was clearly not what @timebertt and myself had in mind. We still need the prefix (`backup.store.prefix`) to be immutable, as was requested in the original issue - just not the entire `backup.store`.